src/odemis/model/_futures.py

```diff
@@ -502,21 +502,29 @@ class ProgressiveBatchFuture(ProgressiveFuture):
     """
     def __init__(self, futures):
         """
-        futures (ProgressiveFuture --> float): dict specifying futures and estimated times
+        :param futures (dict: ProgressiveFuture --> float): Keys are futures and values are the
+                        respective time estimates for the duration of the future.
         """
         self.futures = futures
-        start = time.time()
+        start = time.time()  # start time of the batch future
         super().__init__(start=start, end=start + sum(self.futures.values()))
         self.set_running_or_notify_cancel()
 
         for f in self.futures:
-            f.add_update_callback(self._on_future_update)
-            f.add_done_callback(self._on_future_done)
+            f.add_update_callback(self._on_future_update)  # called whenever set_progress is called
```

Comment:
is this when set_progress is called for the single future, or for the batch future?

Reply:
I think on the single future. Remember that the breakpoint in the `_on_future_update` was hit after every tile. And then `set_progress` on the batch future is called there.

Reply:
My expectation is that this is called whenever the set_progress of each "single" future is called. That might need clarification in the comment.